File: ios-native/Dimo/App/AppConfig.swift
import Foundation

enum AppConfig {
  static var convexURL: String {
    Bundle.main.object(forInfoDictionaryKey: "ConvexURL") as? String ?? ""
  }

  static var workOSClientID: String {
    Bundle.main.object(forInfoDictionaryKey: "WorkOSClientID") as? String ?? ""
  }

  static var gmailOAuthClientID: String {
    Bundle.main.object(forInfoDictionaryKey: "GmailOAuthClientID") as? String ?? ""
  }

  static var gmailOAuthRedirectScheme: String {
    Bundle.main.object(forInfoDictionaryKey: "GmailOAuthRedirectScheme") as? String ?? ""
  }

  static var gmailOAuthRedirectURI: String {
    "\(gmailOAuthRedirectScheme):/oauthredirect"
  }

  static var isGmailConfigured: Bool {
    !gmailOAuthClientID.isEmpty
      && !gmailOAuthRedirectScheme.isEmpty
      && !gmailOAuthClientID.contains("REPLACE_ME")
      && !gmailOAuthRedirectScheme.contains("REPLACE_ME")
      && !gmailOAuthClientID.contains("$(")
      && !gmailOAuthRedirectScheme.contains("$(")
  }

  static let workOSRedirectURI = "dimo://callback"
  static let workOSAuthBaseURL = "https://api.workos.com"
  static let workspaceID = "global"
  static let emailRefreshTaskIdentifier = "app.dimo.ios.email-refresh"
  static let emailAnalysisTaskIdentifier = "app.dimo.ios.email-analysis"

  static var isConfigured: Bool {
    !convexURL.isEmpty && !workOSClientID.isEmpty
      && !convexURL.contains("$(") && !workOSClientID.contains("$(")
  }
}

File: android-native/app/src/main/java/app/dimo/android/app/AppConfig.kt
package app.dimo.android.app

import app.dimo.android.BuildConfig

/**
 * Runtime configuration, the Android counterpart of `ios-native/Dimo/App/AppConfig.swift`.
 * iOS reads these from `Info.plist` (fed by the xcconfig files under `Config/`);
 * Android reads them from `BuildConfig` fields set per product flavor.
 */
object AppConfig {
  val convexURL: String = BuildConfig.CONVEX_URL
  val workOSClientID: String = BuildConfig.WORKOS_CLIENT_ID
  val workOSRedirectURI: String = BuildConfig.WORKOS_REDIRECT_URI
  val workOSAuthBaseURL: String = BuildConfig.WORKOS_AUTH_BASE_URL

  val isConfigured: Boolean
    get() = convexURL.isNotBlank() && workOSClientID.isNotBlank()

  // MARK: - Gmail

  val gmailOAuthClientID: String = BuildConfig.GMAIL_OAUTH_CLIENT_ID
  val gmailOAuthRedirectScheme: String = BuildConfig.GMAIL_OAUTH_REDIRECT_SCHEME
  val gmailOAuthRedirectURI: String get() = "$gmailOAuthRedirectScheme:/oauthredirect"

  /**
   * False until a Google Cloud OAuth client for this package + signing key is put
   * in `android-native/gmail.properties`. The Email tab checks this before
   * offering to connect an inbox.
   */
  val isGmailConfigured: Boolean
    get() = gmailOAuthClientID.isNotBlank() &&
      gmailOAuthRedirectScheme.isNotBlank() &&
      !gmailOAuthClientID.contains("REPLACE_ME") &&
      !gmailOAuthRedirectScheme.contains("REPLACE_ME")
}
